pymc3/model.py

```diff
@@ -1695,16 +1696,30 @@ def pandas_to_array(data):
     XXX: When `data` is a generator, this will return a Theano tensor!
 
     """
-    if hasattr(data, "values"):  # pandas
-        if data.isnull().any().any():  # missing values
-            ret = np.ma.MaskedArray(data.values, data.isnull().values)
+    if hasattr(data, "to_numpy"):
+        vals = data.to_numpy()
+        mask = np.isnan(vals)
+        if mask.any():
+            # there are missing values
+            ret = np.ma.MaskedArray(vals, mask)
```

Comment:
why not just keep it as it is but with `to_numpy` instead of `values`, so that `pandas`' own `isnull` is used? If you're explicitly checking for `to_numpy` (rather than just `PandasObject) then `isnull` will be present
```suggestion
        vals = data.to_numpy()
        mask = data.isnull().to_numpy()
        if mask.any():
            # there are missing values
            ret = np.ma.MaskedArray(vals, mask)
```

Reply:
otherwise you risk
```python
>>> s = pd.Series([1, 3, pd.NA])
>>> s.to_numpy()
array([1, 3, <NA>], dtype=object)
>>> np.isnan(s.to_numpy())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```
instead of
```python
>>> s.isnull().to_numpy()
array([False, False,  True])
```

Reply:
By just checking for `to_numpy` I'd have to trust pandas about `isnull` being there ;)
Also, the `any().any()` was a bit weird and this way anything that can do `.to_numpy()` works. As soon as we get the `ndarray` we're fine.

Reply:
Oh sorry, I did not notice your second comment. I never used `pd.NA` and don't really see the benefit over `float('nan')`. Not sure to what extend we want to support such `dtype=object` inputs.